rusk/src/lib/services/prover/stco.rs

```diff
@@ -0,0 +1,116 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use super::*;
+
+const STCO_INPUT_LEN: usize = u64::SIZE
+    + JubJubScalar::SIZE
+    + JubJubScalar::SIZE
+    + u64::SIZE
+    + PublicSpendKey::SIZE
+    + JubJubAffine::SIZE
+    + Message::SIZE
+    + JubJubScalar::SIZE
+    + Crossover::SIZE
+    + Fee::SIZE
+    + BlsScalar::SIZE
+    + Signature::SIZE;
+
+pub static STCO_PROVER_KEY: Lazy<ProverKey> = Lazy::new(|| {
+    let keys = keys_for(&SendToContractObfuscatedCircuit::CIRCUIT_ID)
+        .expect("keys to be available");
+    let pk = keys.get_prover().expect("prover to be available");
+    ProverKey::from_slice(&pk).expect("prover key to be valid")
+});
+
+impl RuskProver {
+    pub(crate) fn prove_stco(
+        &self,
+        request: &StcoProverRequest,
+    ) -> Result<Response<StcoProverResponse>, Status> {
+        let mut reader = &request.circuit_inputs[..];
+
+        if reader.len() != STCO_INPUT_LEN {
```

Comment:
As discussed, this whole deser could be in its own method so that we can remove all the `map_err` and using `?` to convert from `dusk_bytes::Error` to Rusk's Error to `Status`.